Adding `accept_license` to Chef Analytics

mengesb opened the pull request:
### Description

Addresses an issue with `api_fqdn` by using a helper to ensure it is defined and also fixes Chef's recent requirement to accept the MLSA on reconfigure. Added Ubuntu 14.04 and CentOS 7.2 to the Kitchen test suite - tests are currently passing.
### Issues Resolved

Issue #5 
### Check List
- [x] All tests pass. See https://github.com/chef-cookbooks/community_cookbook_documentation/blob/master/TESTING.MD
- [x] New functionality includes testing.
- [x] New functionality has been documented in the README if applicable
- [x] The CLA has been signed. See https://github.com/chef-cookbooks/community_cookbook_documentation/blob/master/CONTRIBUTING.MD

mengesb committed:
Adding `accept_license` to Chef Analytics

binamov commented on recipes/default.rb:
```diff
@@ -14,10 +14,18 @@
   channel node['chef-analytics']['channel'].to_sym
   version node['chef-analytics']['version']
   package_source node['chef-analytics']['package_source']
+  accept_license node['chef-analytics']['accept_license'] unless node['chef-analytics']['accept_license'].nil?
```
yes, please

lamont-granquist commented on libraries/helpers.rb:
```diff
@@ -0,0 +1,22 @@
+    def api_fqdn_available?
+      return false if node['chef-analytics'].nil?
+      return false if node['chef-analytics']['api_fqdn'].nil?
+      !node['chef-analytics']['api_fqdn'].empty?
```
`node['chef-analytics'] && node['chef-analytics']['api_fqdn'] && !node['chef-analytics']['api_fqdn'].empty?`

its a long expression, but until we can drop support for 2.1+2.2 and can use the new operator in 2.3 that is more idiomatic for protecting against method chaining trainwrecks.

mengesb replied:
At present I don't think we can rely on 2.3 in the DK yet.

lamont-granquist replied:
Right, but its more idiomatic to use the expression I posted there until 2.3 ships --  the `return false` takes me longer to read the code because i wind up reading that and making certain all the edge conditions match the `&&` chaining version or not.

lamont-granquist commented on recipes/default.rb:
```diff
@@ -14,10 +14,18 @@
+  block { repair_api_fqdn }
+  only_if { api_fqdn_available? }
+  not_if { api_fqdn_resolves? }
+end
```
this is going to create a race-to-lazy-everything.

any future resource that uses `node['chef-analytics']['api_fqdn']` must now reference it as `lazy { node['chef-analytics']['api_fqdn'] }` and no resource that comes before this in the run_list can possibly get the correct value.

if the attempt is to lazy this so that it can repair the value from a following cookbook which sets it in compile mode, then the appropriate fix for that issue is that the subsequent cookbook should set this value in its attribute file (that cookbook should be more eager instead of making this cookbook more lazy).

mengesb replied:
As noted in @binamov PR, this was pulled in from https://github.com/chef-cookbooks/chef-server/blob/master/libraries/helpers.rb.

There were notes in that the FQDN is rather important to Analytics, so I thought it prudent to include it in this PR to make the cookbook more resilient, however indeed I do prefer to make it as explicit as possible we need this set. That block is intended to ensure we're setting it or applying the CYA policy.

lamont-granquist replied:
Okay I was reading it wrong yesterday, its not mutating the attribute at all its just creating a file_edit resource out of a ruby_block and Chef::Utils::FileEdit.  That is also bad, but not on the scale of what I thought I was reading yesterday, and I don't have any better fixes.

mengesb committed:
Removed helpers.rb and ruby_block
- Feedback from @binamov and @lamont-grinquist taken into account
- Submitting seperate PR to see if the repo owner wants the helpers.rb (per chef-server cookbook style)

mengesb commented:
OK.

I've removed the `helpers.rb` file and `ruby_block` code referencing it. This should be a pretty clean change (docs and accept_license attribute).

I'll submit a subsequent PR or open an issue to see if the `helpers.rb` is even wanted.

mengesb commented:
Haha, sorry, i mis-tagged your name @lamont-granquist. Thanks for your comments as well!

binamov commented:
👍  🚀
